src/vs/editor/common/config/editorOptions.ts

```diff
@@ -1462,6 +1462,43 @@ class EditorFontSize extends SimpleEditorOption<EditorOption.fontSize, number> {
 
 //#endregion
 
+//#region fontWeight
+
+class EditorFontWeight extends BaseEditorOption<EditorOption.fontWeight, string> {
+	private static ENUM_VALUES = ['normal', 'bold', '100', '200', '300', '400', '500', '600', '700', '800', '900'];
+	private static MINIMUM_VALUE = 1;
+	private static MAXIMUM_VALUE = 1000;
+
+	constructor() {
+		super(
+			EditorOption.fontWeight, 'fontWeight', EDITOR_FONT_DEFAULTS.fontWeight,
+			{
+				anyOf: [
+					{
+						type: 'number',
+						minimum: EditorFontWeight.MINIMUM_VALUE,
+						maximum: EditorFontWeight.MAXIMUM_VALUE
+					},
+					{
+						enum: EditorFontWeight.ENUM_VALUES
+					}
+				],
+				default: EDITOR_FONT_DEFAULTS.fontWeight,
+				description: nls.localize('fontWeight', "Controls the font weight.")
+			}
+		);
+	}
+
+	public validate(input: any): string {
+		if (typeof input === 'number') {
+			return EditorFontWeight.MINIMUM_VALUE <= input && input <= EditorFontWeight.MAXIMUM_VALUE ? String(input) : EDITOR_FONT_DEFAULTS.fontWeight;
```

Comment:
Values outside on `1...1000` range will be marked as invalid and not applied in CSS. Recreated this behavior instead of using `EditorFloatOption.clamp`.